core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -875,6 +880,74 @@ protected ExtractionResult visitInPredicate(InPredicate node, Boolean complement
             return extractionResult;
         }
 
+        private Optional<ExtractionResult> processSimpleInPredicate(InPredicate node, Boolean complement)
+        {
+            if (!(node.getValue() instanceof SymbolReference)) {
+                return Optional.empty();
+            }
+            Symbol symbol = Symbol.from(node.getValue());
+            Map<NodeRef<Expression>, Type> expressionTypes = analyzeExpression(node);
+            Type type = expressionTypes.get(NodeRef.of(node.getValue()));
+            InListExpression valueList = (InListExpression) node.getValueList();
+            List<Object> inValues = new ArrayList<>(valueList.getValues().size());
+            List<Expression> excludedExpressions = new ArrayList<>();
+
+            for (Expression expression : valueList.getValues()) {
+                Object value = new ExpressionInterpreter(expression, metadata, session, expressionTypes)
```

Comment:
This should normally be handled by other optimizers. Why is it necessary to do it here?

Reply:
It's necessary to obtain the native value from a constant expression.
optimization is others' responsibility, but we need to handle whatever LiteralEncoder constructs, for a type.

(btw we use this pattern in other places, for same reason.)